nanomq/pub_handler.c

```diff
@@ -84,25 +47,28 @@ foreach_client(
 	packet_subscribe * sub_pkt;
 	struct client_ctx *ctx;
 
-	for (int i = 0; i < cvector_size(cli_ctx_list); i++) {
-		equal   = false;
+	int      ctx_list_len = cvector_size(cli_ctx_list);
+	uint32_t pids[ctx_list_len];
+
+	for (int i = 0; i < ctx_list_len; i++) {
 		ctx     = (struct client_ctx *) cli_ctx_list[i];
+		pids[i] = ctx->pid.id;
```

Comment:
still risky to use ctx pointer here. better let lib guarantee the ctx_list is threaded safe.